Updating Savings Wallet with PIN explanation (#979)
BitcoinDesign/Guide: guide/savings-wallet.md

### Signing device key replacement

Earlier we mentioned that one of the keys is saved on the user's mobile device and backed up to the cloud. To prevent unauthorized access to this backup, a PIN is created during wallet set-up and used to encrypt and decrypt the backup.

> If a user loses their mobile device but still has their cloud backup and signing device, they can use their PIN to unlock the cloud key. Then, with the help of the signing device, they can recover their wallet.

Reviewer: ```suggestion
If a user loses their mobile device but made a cloud backup, they can use their PIN to unlock the cloud key and restore on a new device.
```

Author: Are you implying that the fact that you made the backup, that you still have access to it? I was trying to make it more explicit by saying that you also still need to have access to it. 

I also added the second key you would need so that the focus is more on how the PIN works in the bigger context of all the keys you need to recover/replace.

Reviewer: In this context I think it is pretty clear that they have access to the cloud backup (the reference design saves it to iCloud or Google Drive, so if they made it, it is there). 

The signing device key does nothing to recover the wallet in this case. Simply restoring from the cloud backup is enough.

Author: Right, but saving it does not mean that you have access to it. If you lost your cloud account sign-in details or manually delete the backup, you lose access to that backup. I'm okay with not making it explicit here and leaving it for a Scenarios section where we explain it in more detail.

Also, to make sure I understand your second point correctly if someone accesses your cloud and knows your PIN, they will be able to restore the wallet on any device and send bitcoin? Obviously, until the max send limit is reached and the signing device is required?

Reviewer: **Re: saving / retrieving the backup**

Yes, if the user loses access to their iCloud or Google account, they will not be able to retrieve the file. For the reference design, that is not the critical point. As this is a multi-sig design, the user will be able to replace the missing key (caveats apply, they need access to the two others of course).

From a developer perspective, what the reference design does here is save the backup to a dedicated part of the cloud service that is only for data from that app, it is not accessible to the user if they 'go look' in their cloud folder.
This is the best option as it protects both from user error (accidentally deleting the file) and malicious access (if someone manages to login to the user's cloud storage, they won't be able to access the file.

**Re: access to funds with backup and PIN**

>> if someone accesses your cloud and knows your PIN, they will be able to restore the wallet on any device and send bitcoin?

No, for two reasons:
1. If the developer uses the approach outlined above (saving the backup to a hidden app specific part of the users cloud), the backup is ONLY available to the app.
2. Even if the bad actor has access to the unencrypted backup, they do not have access to the WALLET as it is a multi-sig setup. They only have one of the three keys. They need two to be able to access the wallet.

Reviewer: After our discussions I think the original change here works fine.
Will resolve this conversation, suggestion can be rejected.

> Given they still have access to their mobile device but they lost their purpose-built signing device, they can replace the lost key (rotate in a new key) with the help of the recovery key held by the auto-signer.

Reviewer: ```suggestion
If they still have access to their mobile device but lost their purpose-built signing device, they can replace the lost key (rotate in a new key) with the help of the recovery key held by the auto-signer.
```
